core/test/mpi/base/bindings.cpp

```diff
@@ -227,7 +229,7 @@ TYPED_TEST(MpiBindings, CanPutValuesWithExclusiveLock)
                 if (rank != my_rank) {
                     win.lock(rank, window::lock_type::exclusive);
                     win.put(this->ref, data.data(), 4, rank, 0, 4);
-                    win.flush(0);
+                    win.flush(rank);
                     win.unlock(rank);
```

Comment:
No unlock does not complete the operation, but just denotes the end of the communication epoch. With a passive target, a flush or flush_all is necessary to enforce the complete the outstanding operations.

Reply:
from https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report/node289.htm, doesn't unlock also guarantee that?
 1-ensure an RMA operation can be completed with UNLOCK at the origin
and then 3-when the RMA is completed at the origin when calling UNLOCK, it's also completed at the target with the same call.

Reply:
Yes, I think for one operation that is valid, but maybe not for multiple operations ? Also, to be honest, I am not sure if all implementations do this properly. The previous bug was partly because of this issue as well. So, a flush is usually a good idea to ensure synchronization.

Reply:
I think since we're only testing the wrapper, IMO it's fine to be overly cautious with semantics, so I would also leave it in.